services/order.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict[str, Any]],
+        username: str,
+        date: str = ""
+) -> Order:
+    with transaction.atomic():
+        order = Order.objects.create(
+            user=get_user_model().objects.get(username=username),
```

Comment:
use transaction.atomic decorator. The decorator is a more concise and cleaner way to ensure that the function executes within a transaction.